// html-dist/lib/html.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.LINEBREAK_NODE = exports.getDoctype = exports.fromHtml = exports.toHtml = exports.textNode = exports.Node = undefined;

var _vnode = require('virtual-dom/vnode/vnode');

var _vnode2 = _interopRequireDefault(_vnode);

var _vtext = require('virtual-dom/vnode/vtext');

var _vtext2 = _interopRequireDefault(_vtext);

var _vdomToHtml = require('vdom-to-html');

var _vdomToHtml2 = _interopRequireDefault(_vdomToHtml);

var _htmlToVdom = require('html-to-vdom');

var _htmlToVdom2 = _interopRequireDefault(_htmlToVdom);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var textNode = function textNode(text) {
  return new _vtext2.default(text);
};

var getDoctype = function getDoctype(input) {
  var doctypeRegex = /([\s\S]+)<html>/;
  var doctypes = doctypeRegex.exec(input);
  return doctypes ? doctypes[1] : '';
};

var LINEBREAK_NODE = textNode('\n');

var fromHtml = (0, _htmlToVdom2.default)({
  VNode: _vnode2.default,
  VText: _vtext2.default
});

var Node = _vnode2.default;

exports.Node = Node;
exports.textNode = textNode;
exports.toHtml = _vdomToHtml2.default;
exports.fromHtml = fromHtml;
exports.getDoctype = getDoctype;
exports.LINEBREAK_NODE = LINEBREAK_NODE;
