// html-dist/lib/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.setUserArgs = exports.args = exports.run = exports.link = exports.h = exports.googleAnalytics = exports.script = undefined;

var _script = require('./script');

var _script2 = _interopRequireDefault(_script);

var _googleAnalytics = require('./google-analytics');

var _googleAnalytics2 = _interopRequireDefault(_googleAnalytics);

var _link = require('./link');

var _link2 = _interopRequireDefault(_link);

var _run = require('./run');

var _run2 = _interopRequireDefault(_run);

var _h = require('virtual-dom/h');

var _h2 = _interopRequireDefault(_h);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// called from the CLI to allow custom arguments being passed
var args = {};
function setUserArgs(newArgs) {
  exports.args = args = newArgs;
}

exports.script = _script2.default;
exports.googleAnalytics = _googleAnalytics2.default;
exports.h = _h2.default;
exports.link = _link2.default;
exports.run = _run2.default;
exports.args = args;
exports.setUserArgs = setUserArgs;
